File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min = float('inf')
        prof = 0

        for i in prices:
            if i < min:
                min = i
            if i - min > prof:
                prof = i-min
        return prof

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max_prof = 0;
        int min_day = 0;
        for(int i = 0; i < prices.length; i++){
            min_day = prices[min_day]<prices[i] ? min_day:i;
            max_prof = max_prof > prices[i] - prices[min_day] ? max_prof: prices[i] - prices[min_day];
        }
        return max_prof;
    }
}